### PR TITLE
Fix "Author" field

### PR DESCRIPTION
**Pull request by KassFlute**
All 'author' fields are replaced by 'owner'. The new 'owner' field is then made clickable to redirect to the corresponding profile page

Tests were updated for this new functionality

### CODE REVIEW

**Review by Mass-14 — CHANGES_REQUESTED** [review]
Don't worry for the coverage, this will be fixed with the new App.kt test, these low scores are only a consequence of SonarCloud illogical way to count the coverage :( .
Can you please add javadocs on public functions of UserViewModel, we have very close functions hence it is necessary.

The renaming Author to Owner is great, however the functionality of going to the account screen is not working when I am not the owner.

What behaviour I see is the following : 
1. click on an item on the loan screen
2. click on the owner's name 
3. I am redirected to my account the owner's data for few frames (like screenshot below) before it reloads and takes my data.

The behaviour at this point should be the following : 
- I am redirected to a new screen with the owner's data, without the _edit profile_ button. 
- The top level destination of this screen should not be "account" since it is not my account
 
image

File: app/src/main/java/com/android/partagix/model/UserViewModel.kt [hunk 1]
```diff
@@ -61,6 +61,10 @@ class UserViewModel(
         )
   }
 
+  fun setUser(user: User) {
+    updateUIState(user)
+  }
```
Comment (Patrick-Pataky) [inline app/src/main/java/com/android/partagix/model/UserViewModel.kt:66]: To easily improve coverage, you can add a test for this function (should be very easy to test)

**Review by Patrick-Pataky — CHANGES_REQUESTED** [review]
Indeed, 50% of coverage is low. Part is due to missing tests in App.kt, but also missing tests for the `UserViewModel`.

I think you can easily add tests for the untested functions in this view model. Indeed, your PR is strongly connected to the user view model, so it is not irrelevant to add them.
- Note: the untested functions are really small & simple. Adding the tests won't take you more than 20min ^^

**Comment by KassFlute** [comment]
I said in my comment that I'll test the view model @Patrick-Pataky.
@Mass-14 For mysterious reasons when I tested myself to look at other's profile it was working but I see why it couldn't work.  And anyways, it makes way more sense to have a different screen to display other's profile. It also makes sense to have a second viewModel for this new screen. I did this in the last commits that you can check if you want. I'll ask for review after adding tests for this.

**Review by Mass-14 — CHANGES_REQUESTED** [review]
image
Remarks : 
- the text in the top bar should be fixed to one line (you can use `maxLines = 1,` and  `overflow = TextOverflow.Ellipsis,` as Text arguments.
- There should be a big spacer (min 20.dp) at on top of the image 
- As well as a small padding (~8.dp) below the image.

image

**Comment by KassFlute** [comment]
Thanks for your review, I implemented your good suggestions.

**Review by Patrick-Pataky — APPROVED** [review]
It seems that you obtained enough coverage. Good job

**Review by Mass-14 — APPROVED** [review]
Great work :)

**Pull request merged**